### PR TITLE
[SPARK-17442][SPARKR] Additional arguments in write.df are not passed to data source

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

additional options were not passed down in write.df.
## How was this patch tested?

unit tests
@falaki @shivaram

### CODE REVIEW

File: R/pkg/inst/tests/testthat/test_sparkSQL.R [hunk 1]
```diff
@@ -243,7 +243,15 @@ test_that("read csv as DataFrame", {
   expect_equal(count(withoutna2), 3)
   expect_equal(count(where(withoutna2, withoutna2$make == "Dummy")), 0)
 
+  # writing csv file
+  csvPath2 <- tempfile(pattern = "csvtest2", fileext = ".csv")
+  write.df(df2, path = csvPath2, "csv", header = "true")
+  df3 <- read.df(csvPath2, "csv", header = "true")
```
Comment (shivaram) [inline R/pkg/inst/tests/testthat/test_sparkSQL.R]: We could do this - but I was thinking we could also check if R's `read.csv` is able to read back the file correctly with headers ?

Comment (felixcheung) [inline R/pkg/inst/tests/testthat/test_sparkSQL.R]: interestingly it doesn't work?
```
> read.csv(file = csvPath2)
Error in read.table(file = file, header = header, sep = sep, quote = quote,  :
  no lines available in input
```

looking..

Comment (shivaram) [inline R/pkg/inst/tests/testthat/test_sparkSQL.R]: The path is interpreted as a directory by `write.df`. It then puts in a `part-0000` or a sequence of such files inside the directory

Comment (felixcheung) [inline R/pkg/inst/tests/testthat/test_sparkSQL.R]: right, read.csv doesn't work with wildcard it seems.
testing a fix.

**Comment by shivaram** [comment]
Thanks - the new test looks good

**Comment by shivaram** [comment]
LGTM -  @falaki any more comments ?

**Comment by falaki** [comment]
Hey just saw. This looks good to me too.

**Comment by shivaram** [comment]
Merging into master, branch-2.0

**Pull request closed without merging**